packages/ra-ui-materialui/src/list/Datagrid.tsx

```diff
@@ -197,8 +199,8 @@ const Datagrid: FC<DatagridProps> = props => {
                             />
                         </TableCell>
                     )}
-                    {Children.map(children, (field, index) =>
-                        isValidElement(field) ? (
+                    {Children.map(children, (field, index) => {
+                        return isValidElement(field) ? (
```

Comment:
why this change?

Reply:
I'm reverting this change. It comes from my investigations.